qemu/qemu.go

```diff
@@ -1017,6 +1020,63 @@ func (vsock VSOCKDevice) QemuParams(config *Config) []string {
 	return qemuParams
 }
 
+// RngDevice represents a random number generator device.
+type RngDevice struct {
+	// ID is the device ID
+	ID string
+	// Filename is entropy source on the host
+	Filename string
+	// MaxBytes is the bytes allowed to guest to get from the host’s entropy per period
+	MaxBytes uint
+	// Period is duration of a read period in seconds
+	Period uint
+}
+
+// Valid returns true if the RngDevice structure is valid and complete.
+func (v RngDevice) Valid() bool {
+	if v.ID == "" {
```

Comment:
Is `ID` the only thing we need with `virtio-rng` ? Will Qemu fill up the optional field (Filename, MaxBytes, Period) ?

Reply:
@sboeuf yes qemu will fill up the fields, actually even ID is optional you cant only do `-device virtio-rng-pci` but I followed the govmm basic checks (that make sense for control of devices that are added).

Reply:
ok makes sense!